### DISCUSSION TITLE
Better script to migrate existing partitions to partman?

### DISCUSSION THREAD

**Discussion by Faithfinder** [pgpartman/pg_partman#703] [Q&A]
Current version in the docs parses partition names or data, but there must be a way to read the partition expression. Unfortunately my sql-fu isn't strong enough. Anyone cares to make the improvement:)?

**Comment by keithf4** [comment]
That's a good point. The `show_partition_info` function does contain code for reading the partition expression boundaries.

https://github.com/pgpartman/pg_partman/blob/master/sql/functions/show_partition_info.sql
